modules/auxiliary/scanner/mysql/mysql_authbypass_hashdump.rb

```diff
@@ -103,17 +113,17 @@ def run_host(ip)
           begin
             # Create our socket and make the connection
             s = connect(false)
-            x = ::RbMysql.connect(rhost, username, password, rport, s)
+            mysql_client = ::RbMysql.connect(rhost, username, password, nil, rport, s)
+
             print_good "#{rhost}:#{rport} Successfully bypassed authentication after #{count} attempts. URI: mysql://#{username}:#{password}@#{rhost}:#{rport}"
-            results << x
+            results << mysql_client
           rescue RbMysql::AccessDeniedError
-          rescue Exception => e
+          rescue ::Exception => e
```

Comment:
This was previously only rescue'ing `::Msf::Exception` instead of `::Exception`. After handling `::Exception` correctly, we can see that this error was hidden:

```
[-] 127.0.0.1:3306        - 127.0.0.1:3306 Thread 2] caught an unhandled exception: no implicit conversion of Socket into String
```

Full stacktrace:

```
caught an unhandled exception: no implicit conversion of Socket into String
/Users/user/Documents/code/metasploit-framework/lib/rbmysql/protocol.rb:153:in `initialize'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql/protocol.rb:153:in `new'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql/protocol.rb:153:in `block in initialize'
/Users/user/.rvm/gems/ruby-3.0.5@metasploit-framework/gems/timeout-0.3.2/lib/timeout.rb:173:in `timeout'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql/protocol.rb:147:in `initialize'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql.rb:115:in `new'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql.rb:115:in `connect'
/Users/user/Documents/code/metasploit-framework/lib/rbmysql.rb:50:in `new'
/Users/user/Documents/code/metasploit-framework/modules/auxiliary/scanner/mysql/mysql_authbypass_hashdump.rb:106:in `block in run_host'
```

Which was a regression caused by replacing RbMysql with the latest implementation in this PR:


https://github.com/rapid7/metasploit-framework/pull/14987/files#diff-761f86a38bf96199e26c8070c16d04644f523eb7d3a5315de8818f0ad11d1d43R151-R153

Which meant this manual patch to update RbMysql to support a socket being passed instead of creating a fresh TcpSocket got lost:

https://github.com/dwelch-r7/metasploit-framework/commit/ec9bc73e0fde5caad4c1750f3080b4add0933c71